pontoon/base/static/js/themeSwitcher.js

```diff
@@ -0,0 +1,77 @@
+$(function () {
+  function getSystemTheme() {
+    if (
+      window.matchMedia &&
+      window.matchMedia('(prefers-color-scheme: dark)').matches
+    ) {
+      return 'dark';
+    } else {
+      return 'light';
+    }
+  }
+
+  function updateTheme(newTheme) {
+    if (newTheme === 'system') {
+      newTheme = getSystemTheme();
+    }
+    $('body')
+      .removeClass('dark-theme light-theme system-theme')
+      .addClass(`${newTheme}-theme`);
+
+    if (newTheme !== 'system') {
+      Pontoon.endLoader(`Theme changed to ${newTheme}.`);
+    }
+  }
+
+  window
+    .matchMedia('(prefers-color-scheme: dark)')
+    .addEventListener('change', function (e) {
+      if ($('.appearance .toggle-button button.system').hasClass('active')) {
```

Comment:
This will work on the /settings page. But what if another page is opened when the user changes system settings? We could for example store the actual user theme setting in a data-theme attribute on the body element and use that to check if the system theme is enabled.